go-controller/pkg/util/net_linux.go

```diff
@@ -335,26 +335,22 @@ func LinkRoutesAddOrUpdateSourceOrMTU(link netlink.Link, gwIP net.IP, subnets []
 	return nil
 }
 
-// LinkRouteGet gets a route for the given subnet with the specified gwIPStr
+// LinkRouteGetFilteredRoute gets a route for the given route filter.
 // returns nil if route is not found
-func LinkRouteGet(link netlink.Link, gwIP net.IP, subnet *net.IPNet) (*netlink.Route, error) {
-	routeFilter := &netlink.Route{Dst: subnet, LinkIndex: link.Attrs().Index}
-	filterMask := netlink.RT_FILTER_DST | netlink.RT_FILTER_OIF
-	routes, err := netLinkOps.RouteListFiltered(getFamily(gwIP), routeFilter, filterMask)
+func LinkRouteGetFilteredRoute(routeFilter *netlink.Route, filterMask uint64) (*netlink.Route, error) {
+	routes, err := netLinkOps.RouteListFiltered(getFamily(routeFilter.Dst.IP), routeFilter, filterMask)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get routes for subnet %s", subnet.String())
+		return nil, fmt.Errorf("failed to get routes for subnet %s", routeFilter.Dst.String())
```

Comment:
Should we use the filter details in the error message instead?
Meaning, when we see this error, will it be easy to discern which usage of `LinkRouteGetFilteredRoute` of the two failed? Ignore if it is.

Reply:
Just clarity, you want to also print the `filterMask` @jcaamano ?

Reply:
Yeah, that would not do any harm

Reply:
Done.